twitter/View Controllers/TweetDetailsViewController.m

```diff
@@ -55,8 +59,79 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 */
 
 - (IBAction)didTapFavorite:(id)sender {
+    // Update the local tweet model
+    if (self.tweet.favorited) { // user is unliking the tweet
+        self.tweet.favoriteCount--;
+        self.tweet.favorited = false;
+        // Update cell UI
+        [self refreshFavoritesData];
+        // Send a POST request to the POST unfavorite endpoint
+        [[APIManager shared] unfavorite:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if (error) {
+                  NSLog(@"Error unfavoriting tweet: %@", error.localizedDescription);
```

Comment:
Would we want to revert the favourited / unfavourited tweet in the case of an error, so that the local change matches the backend?

Reply:
Thanks pointing this out! Just added lines to revert any increment/decrement of counts in case of error.